specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/storageTaskAssignments.json

```diff
@@ -0,0 +1,842 @@
+        },
+        "report": {
+          "$ref": "#/definitions/StorageTaskAssignmentReport",
+          "description": "The storage task assignment report"
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Represents the provisioning state of the storage task assignment.",
+          "enum": [
+            "ValidateSubscriptionQuotaBegin",
+            "ValidateSubscriptionQuotaEnd",
+            "Creating",
+            "Succeeded",
+            "Deleting",
+            "Canceled",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": false
+          }
+        },
+        "runStatus": {
+          "$ref": "./common.json#/definitions/StorageTaskReportProperties",
+          "description": "Run status of storage task assignment"
+        }
+      },
+      "required": [
+        "taskId",
+        "enabled",
+        "description",
+        "executionContext",
+        "report"
+      ],
+      "description": "Properties of the storage task assignment."
+    },
+    "StorageTaskAssignmentExecutionContext": {
+      "type": "object",
+      "properties": {
+        "target": {
+          "$ref": "#/definitions/ExecutionTarget",
+          "description": "Execution target of the storage task assignment"
+        },
+        "trigger": {
+          "$ref": "#/definitions/ExecutionTrigger",
+          "description": "Execution trigger of the storage task assignment"
+        },
+        "constraints": {
+          "$ref": "#/definitions/ExecutionConstraints",
+          "description": "Execution constraints of the storage task assignment"
+        }
+      },
+      "required": [
+        "target",
+        "trigger"
+      ],
+      "description": "Execution context of the storage task assignment."
+    },
+    "ExecutionTarget": {
+      "type": "object",
+      "properties": {
+        "prefix": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of target prefixes"
+        },
+        "excludePrefix": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of target prefixes to exclude"
+        },
+        "reservedPathPrefix": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of reserved path prefixes"
+        }
+      },
+      "required": [
+        "prefix"
+      ],
+      "description": "Target for storage task assignment execution"
+    },
+    "ExecutionTrigger": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "RunOnce",
+            "OnSchedule"
+          ],
+          "x-ms-enum": {
+            "name": "TriggerType",
+            "modelAsString": false
```

Comment:
>modelAsString [](http://example.com/codeflow?start=13&length=13)

SDK folks, is modelAsString false a good idea? I was under the impression its not really recommended, but I have trouble thinking of why.

Reply:
Is it just false in this set of APIs for consistency?

Reply:
Yes we are just following the pattern used in for other APIs in Microsoft.Storage - We would ideally like to keep them untouched unless there is any rationale behind not modeling them as string.

Also, in favor of getting the sign off faster, given the PR is open for weeks with multiple reviewers, we would make the change if that allows you to close this comment faster tim

Reply:
OK, don't worry about it for this API.